test/spec/test_spec.rb

```diff
@@ -0,0 +1,67 @@
+require_relative "spec_helper"
+
+describe "A PHP application" do
+
+  it "works with the getting started guide" do
+    Hatchet::Runner.new("php-getting-started").tap do |app|
+      app.deploy do
+      # deploy works
+      end
+    end
+  end
+
+  it "checks for bad version" do
+    Hatchet::Runner.new("php-getting-started", allow_failure: true).tap do |app|
+      app.before_deploy do
+        File.open("composer.json", "w+") do |f|
+          f.write '{
+          "require": {
+            "php": "7.badversion"
+          }}'
+        end
+      end
+      app.deploy do
+        expect(app.output).to include("Could not parse version constraint 7.badversion")
+      end
+    end
+  end
+
+  it "have absolute buildpack paths" do
+    buildpacks = [
+      :default,
+      "https://github.com/sharpstone/force_absolute_paths_buildpack"
+    ]
+    Hatchet::Runner.new("php-getting-started", buildpacks: buildpacks).deploy do |app|
+      #deploy works
```

Comment:
Does this test fail if the deploy fails if nothing is asserted?

Reply:
Correct. An exception is raised if the deploy fails by default. Here the logic and checks are actually being performed in the buildpack:

https://github.com/sharpstone/force_absolute_paths_buildpack

See the readme for more info